example/metaball.js

```diff
@@ -119,21 +121,92 @@ const drawMetaballs = regl({
 
 const metaball = (px, py, pz, strength, subtract) => {
   return (x, y, z) => {
-    return (strength / (Math.pow(x - px, 2) + Math.pow(y - py, 2) + Math.pow(z - pz, 2))) - subtract
+    let fx = x - px
+    let fy = y - py
+    let fz = z - pz
+    return strength / (0.000001 + (fx * fx) + (fy * fy) + (fz * fz)) - subtract
   }
 }
 
-const numblobs = 20
+const numblobs = 3
 const strength = 1.2 / ((Math.sqrt(numblobs) - 1) / 4 + 1)
 const subtract = 12
-const balls = Array(numblobs).fill().map((_, i) => {
-  return (time) => {
-    let ballx = Math.sin(i + 1.26 * time * (1.03 + 0.5 * Math.cos(0.21 * i))) * 0.27 + 0.5
-    let bally = Math.cos(i + 1.12 * time * 0.21 * Math.sin((0.72 + 0.83 * i))) * 0.27 + 0.5
-    let ballz = Math.cos(i + 1.32 * time * 0.1 * Math.sin((0.92 + 0.53 * i))) * 0.27 + 0.5
-    return metaball(ballx, bally, ballz, strength, subtract)
-  }
-})
+const size = 30
+const bounds = [[0, 0, 0], [1.5, 1.5, 1.5]]
+const position = (time, i) => {
+  return [
+    Math.sin(i + 1.26 * time * (1.03 + 0.5 * Math.cos(0.21 * i))) * 0.27 + 0.5,
+    Math.cos(i + 1.12 * time * 0.21 * Math.sin((0.72 + 0.83 * i))) * 0.27 + 0.5,
+    Math.cos(i + 1.32 * time * 0.1 * Math.sin((0.92 + 0.53 * i))) * 0.27 + 0.5
+  ]
+}
+let start_bounds, end_bounds
+[start_bounds, end_bounds] = bounds
+let step_sizes = [0, 1, 2].map((i) => (end_bounds[i] - start_bounds[i]) / size)
+const r = size * Math.sqrt(strength / subtract)
+
+const render = (tick) => {
+  let time = 0.05 * tick
+  let fieldArray = ndarray(new Float32Array(size * size * size), [size, size, size])
+  let balls = Array(numblobs).fill().map((_, i) => {
+    let bx, by, bz
+    [bx, by, bz] = position(time, i)
+    return {
+      functional: metaball(bx, by, bz, strength, subtract),
+      center: [bx, by, bz]
+    }
+  })
+
+  fill(fieldArray, (i, j, k) => {
+    let val = 0
+
+    for (let n = 0; n < numblobs; n++) {
+      let ball = balls[n]
+      let zs = ball.center[2] * size
+      let ys = ball.center[1] * size
+      let xs = ball.center[0] * size
+
+      var min_z = Math.floor(zs - r)
```

Comment:
I really dislike this code style, but I'm trying to mirror as much as possible the example code. When it looks the same as [this](https://www.clicktorelease.com/code/bumpy-metaballs/) then I'll refactor it to make it more elegant code.